src/popsift/popsift.cpp

```diff
@@ -313,14 +339,21 @@ void PopSift::extractDownloadLoop( )
 
         job->setFeatures( features );
     }
+
+    private_unit();
```

Comment:
Do I understand correct that you want to delete the Pyramid every time you have downloaded the features? That would crash if several images have been queued for feature extraction, wouldn't it?

Reply:
Pyramid is deleted only after pipeline stops using `PopSift::uninit` function. Not that `private_unit` (this is actually a typo and should be `private_uninit`) is called outside while loop.

Reply:
Shouldn't you add private_uninit also to the matchPrepareLoop?

Reply:
Yes, you are right.